tests/fsharp/Compiler/Libraries/Core/LanguagePrimitives/StringFormatTests.fs

```diff
@@ -137,3 +138,70 @@ module ``String Format Tests`` =
         Assert.areEqual (string  infinityf) "Infinity"
         Assert.areEqual (string  nanf) "NaN"
         Assert.areEqual (string (new System.Guid("210f4d6b-cb42-4b09-baa1-f1aa8e59d4b0"))) "210f4d6b-cb42-4b09-baa1-f1aa8e59d4b0"
+
+    [<Test>]
+    let ``string constructor in FSI``() =
+        // Regression test for FSHARP1.0:5894
+
+        CompilerAssert.RunScriptWithOptions [| "--langversion:preview" |]
+            """
+let assertEqual a b =
```

Comment:
Is there some reason you didn't use Assert.areEqual?

Reply:
Partly following the example in https://github.com/dotnet/fsharp/pull/9396#issuecomment-639098947, partly because it seemed easier than importing NUnit into FSI (might well be, that I'm just not aware of an easy way)).

Reply:
Oh of course, the assertion is happening within the executing code, it is not verifying the results.  That makes perfect sense.  Indeed I just did exactly the same thing in a test I added last night.

@vzarytovskii , we should consider having some form of Assertions framework used from within the compiled sources.  That would probably stop us from adding our own unique assertions every time we check-in new tests.

Reply:
> @vzarytovskii , we should consider having some form of Assertions framework used from within the compiled sources. That would probably stop us from adding our own unique assertions every time we check-in new tests.

@KevinRansom, I was pondering the same for a little while. Is there any method to make functions available within such code snippets? Preferably a better one than string manipulation?

Reply:
@ThorstenReichert 
We are running a compiler within the test, so we can simply have a library and ensure the test references it.  That works for the compiler and for FSI.

/cc @vzarytovskii